docs/ADR-58-third-party-assets-integration-v2-with-merkle-tree.md

```diff
@@ -0,0 +1,412 @@
+
+The approval process is performed by the UI as follows:
+
+1. The builder-server is asked for all items that are awaiting curation as the items that have their item curations as `PENDING` have not been approved yet. This is done in a paginated but concurrent way as the amount of items can be big.
+2. The builder-server is asked for the approval data of the collection. This is done by requesting the `GET /collections/{collectionId}/approvalData` endpoint. The server will return the latest signed cheque, a flag to check if the cheque was already consumed or not, a map of items and their item entity hashes and the latest Merkle Tree root. The cheque and the entity hashes were stored in the previous steps in the builder-server database, but the cheque consumption flag and the latest Merkle Tree root used are retrieved from the third party record in the blockchain.
+3. Using the entity hashes, the Merkle Tree is built.
+4. If the Merkle Tree root that was built in the previous step is different from the one returned in the 2nd step from the server, the curator will be asked to perform the approval action against the blockchain. This transaction will contain the third party id, the Merkle Tree root that was computed in the 4th step and the cheque if there are new items to be published. The transaction will store the cheque in the contract and update the Merkle Tree root to allow the deployment of the items from which the tree was made from in the Catalysts.
+5. A polling procedure will be performed to wait for the graph to return the new Merkle Tree root to ensure the next process is performed correctly.
+6. Each item is deployed to the Catalyst by building its entity as explained in the [ADR-62](./ADR-62-merkle-proofed-entities.md) and the item's curation is updated to `APPROVED` in a concurrent manner. The item curation is updated to `APPROVED` only if the item was deployed to the Catalyst.
+
+After the approval process is completed, all the items have their item curation as `APPROVED`, unlocking for further item publications.
+
+#### Catalyst acceptance criteria
+
+Third party items deployed to the Catalyst have a different entity metadata than the standard items. The entity metadata, defined in the [ADR-62](./ADR-62-merkle-proofed-entities.md), is a Merkle Proofed entity metadata, which is an item metadata with an attached Merkle Tree proof to validate that the item to be deployed belongs to a Merkle Tree.
+
+Upon receiving a deployment, the Catalysts will check the graph for the latest Merkle Tree root available:
+
+```GraphQL
+{
+  thirdParties(block:{number : {blockNumber}}) {
```

Comment:
we need to add the `isApproved: true`, right?

Reply:
Done! Added the `isApproved` clause.